### DISCUSSION TITLE
consolidate

### DISCUSSION THREAD

**Discussion by daveceerocker** [rshea0/enjinn#50] [Help]
Hey Ryan

Firstly congratulations on the new app, brilliant.

Ok this is my issue. As you know i have all my tunes on my D: and Tuneup is asking where my Engine Library folder is but i cant choose the path to the D: as it brings up a warning that this isnt a proper Engine Folder. I get that. 

So what i want to do is to be able to choose the D:Tunes folder as my source, and then consolidate the tunes from EnginePrime to a folder in the proper Engine Library. All the tunes that have been prepped in EnginePrime were done from the D:Tunes folder, so i need tuneup to be able to set that D:Tunes as the "Engine Library" if you see what i mean? and then i can consolidate.

ive tried these workarounds though...
1. i created a dump file in the main engine library to put the (to be) consolidated tunes into...
2. i then cut the files from D:Tunes to the Engine Library and then tried to relocate them seeing as they weren't in the original folder anymore (which would make all the files red - which it did when i checked EP)
3. when i try relocate, tuneup finds nothing in the engine library even though all the tunes are now in there now. 

Im fully aware i made a mess of everything by putting the tunes into the D; but i didnt have space on the C; at the time, now ive upgraded to a 2TB drive so i got plenty of room. 

I just want tuneup to find the missing tunes (which are all red) and then stick them in one folder by consolidation.

just to re-cap. all my tunes are now in the proper engine library in the original folder they were all prepped in (the same folder structure, the lot, i cut the whole lot from the D: and even deleted the folder (but of course i can restore that so im not in any bother at the moment but i gave it a try in case having 2 folder structures was confusing tuneup)

Can you help please?

thanks Ryan
Dave

**Comment by shayded-exe** [comment]
So if I understand correctly, your library on D has all your cue points and everything in it, right?

Is your D library a library you exported via the sync manager, or was it a main library you copied to D? This is important because the database format is slightly different for exported libraries.

**Comment by daveceerocker** [comment]
hey

no i didnt export anything, i just started storing tunes and editing them
in Engine Prime from there (instead of C:). There's no database in the
D:Tunes folder. at all.

On Sat, 19 Jun 2021 at 16:03, Ryan Shea ***@***.***> wrote:

> So if I understand correctly, your library on D has all your cue points
> and everything in it, right?
>
> Is your D library an exported library, or was it a main library you copied
> to D? This is important because the database format is slightly different
> for exported libraries.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/tuneup-prime/discussions/50#discussioncomment-891605>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUB2TXOESM6SMJ7JAMBQINDTTSWTPANCNFSM467EFLXQ>
> .
>


-- 
David Crocker
Bering Tees Ltd
+44 (0)7737 269174

**Reply by shayded-exe** [reply]
Ah I see. Could you please email me the `m.db` file from your Engine library to shayded@shayded.com? I want to see exactly what's going on.

All you should have to do though is use consolidate on your C library. When you click consolidate, it asks you where you would like to consolidate to. So if your C library is linked to your files on D, it will work.

**Comment by daveceerocker** [comment]
ive sent you a dropbox request mate. its too big to email.

On Sat, 19 Jun 2021 at 16:35, Ryan Shea ***@***.***> wrote:

> Ah I see. Could you please email me the m.db file from your Engine
> library to ***@***.***? I want to see exactly what's going on.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/tuneup-prime/discussions/50#discussioncomment-891675>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUB2TXLFA2LMUL7NHLFKYWLTTS2KLANCNFSM467EFLXQ>
> .
>


-- 
David Crocker
Bering Tees Ltd
+44 (0)7737 269174

**Reply by shayded-exe** [reply]
Got it. Taking a look.

**Reply by shayded-exe** [reply]
Ahhhhh looks like all your tracks are "external" tracks. Currently tuneup filters these out. I need to add an option to consolidate to import from external libraries (or something like that).

I don't yet fully understand myself how tracks get in this state though. I just added a track on an external drive to my library and Engine didn't add it to my internal library, but instead created a new library on the external drive and added it to that one. Maybe it's something from Engine 1.5.x?

What version of Engine did you use when you added these tracks originally? Also, exactly how did you add them? Did you right click > "add to collection", or did you do something like drag the files in from the file explorer?

**Comment by daveceerocker** [comment]
AHH. Yeh I used v1.5

I've added maybe 10 tunes as 1.6

Thing is is that there's no database outside of the engine library. Or is
there?

Can I do anything more to help you here?

**Reply by shayded-exe** [reply]
I installed Engine 1.5 with a blank library and imported some tunes from an external drive in a multitude of ways, and each time ended up with the same result. It created a library on the external drive and added the tracks to that, not the main library. So it doesn't appear to be a version issue. This was on a mac though. I'll have to try again on my PC.

So there's no "Engine Library" folder on your external drives, yes? (I know you said there's none in your tunes folder, but double check the root of the drive).

Do you remember when you added those tracks, how exactly you did it? Did you right click > "add to collection", or did you do something like drag the files in from the file explorer?

If it isn't too much trouble, try this for me. Start with a blank library (just rename your existing engine library folder, make a backup too if you haven't already), and add some tracks to that library from your external drive. Send me the new `m.db` file. To clarify, use Engine 1.6.

As far as fixing your library goes though, I should be able to use your library file to figure out how to adapt consolidate to work with it. I can send you a pre-release version to test when it's ready.

**Comment by daveceerocker** [comment]
I added these tunes a long time ago by drag n drop. So you're telling me to
check my SSD external device for the m.db yes? I'm not sure how to search
the root of the d drive but from a visual I can't see any files. I can do
what you ask but it'll be tomorrow as I'm off out soon. I think there must
be a dB somewhere or engineprime wouldn't work!

Thanks for your hard work 👍👍

**Reply by shayded-exe** [reply]
It would just be in a folder called "Engine Library" at the root. It's ok if there isn't one. I just wanted to make 100% sure before moving on from that.

You're welcome!

**Comment by daveceerocker** [comment]
Ok I'll do a search on thx external device as well. I'll get back to you
tomorrow 👍👍

**Comment by daveceerocker** [comment]
If I uninstall/reinstall engineprime won't I lose all the data?

**Reply by shayded-exe** [reply]
You don't need to reinstall. And no, all your data is stored in your Engine Library folder.

**Comment by daveceerocker** [comment]
Hey Ryan

Ok please check the dropbox folder again. There *were* data files on my ssd
device as well, so ive zipped them up to the drop box and also i created a
new crate in EP Library and dropped 3 tunes there, this time its the proper
Engine Library on the C: drive

I hope the files help

thanks
Dave

On Sat, 19 Jun 2021 at 19:34, Ryan Shea ***@***.***> wrote:

> It would just be in a folder called "Engine Library" at the root. It's ok
> if there isn't one. I just wanted to make 100% sure before moving on from
> that.
>
> You're welcome!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/tuneup-prime/discussions/50#discussioncomment-892893>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUB2TXN4T7RSVGH3VNT5WPLTTTPLFANCNFSM467EFLXQ>
> .
>


-- 
David Crocker
Bering Tees Ltd
+44 (0)7737 269174

**Reply by shayded-exe** [reply]
Thanks, I'll dig through when I have a chance

**Comment by daveceerocker** [comment]
Yes there was an Engine Library folder on my external SSD (i zipped the
data base files up for you) i didnt create this though.

thanks

On Sat, 19 Jun 2021 at 19:01, Ryan Shea ***@***.***> wrote:

> I installed Engine 1.5 with a blank library and imported some tunes from
> an external drive in a multitude of ways, and each time ended up with the
> same result. It created a library on the external drive and added the
> tracks to that, not the main library. So it doesn't appear to be a version
> issue. This was on a mac though. I'll have to try again on my PC.
>
> So there's no "Engine Library" folder on your external drives, yes? (I
> know you said there's none in your tunes folder, but double check the root
> of the drive).
>
> Do you remember when you added those tracks, how exactly you did it? Did
> you right click > "add to collection", or did you do something like drag
> the files in from the file explorer?
>
> If it isn't too much trouble, try this for me. Start with a blank library
> (just rename your existing engine library folder, make a backup too if you
> haven't already), and add some tracks to that library from your external
> drive. Send me the new m.db file. To clarify, use Engine 1.6.
>
> As far as fixing your library goes though, I should be able to use your
> library file to figure out how to adapt consolidate to work with it. I can
> send you a pre-release version to test when it's ready.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/tuneup-prime/discussions/50#discussioncomment-892822>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUB2TXOWW2PMLE7OLNOREWTTTTLPZANCNFSM467EFLXQ>
> .
>


-- 
David Crocker
Bering Tees Ltd
+44 (0)7737 269174

**Reply by shayded-exe** [reply]
Hey just following up. I've started updating the app again. Apologies for the long absence.

Do you still need help with this or have you sorted it out?

**Comment by daveceerocker** [comment]
Hi. Gave upon Denon and went to pioneer. Good luck 👍👍
